### PR TITLE
Adding collection for ansible-galaxy

### PR DESCRIPTION
**Pull request by rohit-dimagi**
##### SUMMARY

* Adding collection for ansible-galaxy

##### ENVIRONMENTS AFFECTED

* ALL
##### ISSUE TYPE

- Change Pull Request

### CODE REVIEW

File: src/commcare_cloud/ansible/requirements.yml [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
+  - src: git+https://github.com/mrlesmithjr/ansible-squid.git
+    version: 2d015170b7e4f28e4bc782af829f340974efaf2d
+    name: squid
+  - src: git+https://github.com/nickhammond/ansible-logrotate.git
+    version: e7a498
+    name: ansible-logrotate
+  - src: DavidWittman.redis
+    version: 1.2.5
+collections:
+  - name: https://github.com/dimagi/commcare-logstash/releases/download/V0.1/dimagi-commcare-logstash-1.0.0.tar.gz
```
Comment (snopoke) [inline src/commcare_cloud/ansible/requirements.yml]: how come the tag is `V0.1` but the tar is `1.0.0`?

File: src/commcare_cloud/manage_commcare_cloud/install.py [hunk 2]
```diff
@@ -23,13 +23,20 @@ def run(self, args, unknown_args):
...
-        cmd_parts = ['ansible-galaxy', 'install', '-f', '-r', os.path.join(ANSIBLE_DIR, 'requirements.yml')]
-        cmd = ' '.join(shlex_quote(arg) for arg in cmd_parts)
-        print_command(cmd)
-        p = subprocess.Popen(cmd, stdin=subprocess.PIPE, shell=True, env=env)
-        p.communicate()
+        env['ANSIBLE_COLLECTION_PATHS'] = ANSIBLE_COLLECTION_PATHS
+        cmd_roles_parts = ['ansible-galaxy', 'install', '-f', '-r', os.path.join(ANSIBLE_DIR, 'requirements.yml')]
+        cmd_collection_parts = ['ansible-galaxy', 'collection', 'install', '-f', '-r', os.path.join(ANSIBLE_DIR, 'requirements.yml'),  '-p', ANSIBLE_COLLECTION_PATHS]
+        
+        for cmd_parts in (cmd_roles_parts,cmd_collection_parts):
```
Comment (snopoke) [inline src/commcare_cloud/manage_commcare_cloud/install.py]:
```suggestion
        for cmd_parts in (cmd_roles_parts, cmd_collection_parts):
```

File: src/commcare_cloud/manage_commcare_cloud/install.py [hunk 3]
```diff
@@ -23,13 +23,20 @@ def run(self, args, unknown_args):
...
+            os.makedirs(ANSIBLE_COLLECTION_PATHS)
 
         env['ANSIBLE_ROLES_PATH'] = ANSIBLE_ROLES_PATH
-        cmd_parts = ['ansible-galaxy', 'install', '-f', '-r', os.path.join(ANSIBLE_DIR, 'requirements.yml')]
-        cmd = ' '.join(shlex_quote(arg) for arg in cmd_parts)
-        print_command(cmd)
-        p = subprocess.Popen(cmd, stdin=subprocess.PIPE, shell=True, env=env)
-        p.communicate()
+        env['ANSIBLE_COLLECTION_PATHS'] = ANSIBLE_COLLECTION_PATHS
+        cmd_roles_parts = ['ansible-galaxy', 'install', '-f', '-r', os.path.join(ANSIBLE_DIR, 'requirements.yml')]
```
Comment (snopoke) [inline src/commcare_cloud/manage_commcare_cloud/install.py]: Just noticed this. I think you have to specify `role` here to install the roles:
```suggestion
        cmd_roles_parts = ['ansible-galaxy', 'role', 'install', '-f', '-r', os.path.join(ANSIBLE_DIR, 'requirements.yml')]
```

From the docs:
> While both roles and collections can be specified in one requirements file, they need to be installed separately. The ansible-galaxy role install -r requirements.yml will only install roles and ansible-galaxy collection install -r requirements.yml -p ./ will only install collections.

**Comment by rohit-dimagi** [comment]
Should we hold onto this? ansible 2.9 ran into some issue( which i know of) there could be others.

File: src/commcare_cloud/ansible/requirements.yml (hunk 1 above)
Comment (rohit-dimagi) [reply src/commcare_cloud/ansible/requirements.yml]: Updated tag

**Pull request merged**